merlin/models/tf/utils/tf_utils.py

```diff
@@ -248,3 +248,39 @@ def add_epsilon_to_zeros(tensor: tf.Tensor, epsilon: float = 1e-24) -> tf.Tensor
         The tensor without zeros
     """
     return tf.where(tf.equal(tensor, 0.0), tensor + epsilon, tensor)
+
+
+class TensorInitializer(tf.keras.initializers.Initializer):
+    """Initializer that returns a tensor (e.g. pre-trained
+    embeddings) set in the constructor
+    """
+
+    def __init__(self, weights: Union[tf.Tensor, Any]):
+        self._weights = tf.convert_to_tensor(weights)
+
+    def __call__(self, shape: tf.TensorShape, dtype: tf.DType = None, **kwargs) -> tf.Tensor:
+        """Returns a tensor object initialized with the tensor
+        set in the constructor.
+
+        Parameters
+        ----------
+        shape : tf.TensorShape
+            Shape of the variable to be initialized
+        dtype : tf.DType, optional
+            Optional dtype of the tensor. Only numeric or boolean dtypes are
+        supported, by default None
+
+        Returns
+        -------
+        tf.Tensor
+            Returns the tensor set in the constructor
+        """
+        tf.assert_equal(shape, self._weights.shape)
+
+        weights = self._weights
+        if dtype:
+            weights = tf.cast(self._weights, dtype)
+        return weights
+
```

Comment:
Would be good to add a class-method `from_dataset` where we convert a merlin.io.Dataset to the weights.

Reply:
I have added `TensorInitializer.from_dataset()` as you suggested

Reply:
@marcromeyn I have added `TensorInitializer.from_dataset()` as you suggested. 
But end up finding a bug #406, as I noticed that `cudf.from_dlpack()` don't work properly with DLpack encoded 2D TF tensors. The fix of that issue depends on the issue [#10754](https://github.com/rapidsai/cudf/issues/10754) I created on `cudf` repo.
But I think we can merge this PR as `TensorInitializer.from_dataset()` is about loading the embeddings from a dataset, an the issue occurs when exporting the embeddings.